_docs/tasks/istio-auth.md

```diff
@@ -9,166 +9,157 @@ type: markdown
+The following commands assume the services are deployed in the default namespace.
+Use the parameter *-n yournamespace* to specify a namespace other than the default one.
 
-The following command removes Istio CA and its namespace *istio-system*.
+1. Verify AuthPolicy setting in ConfigMap.
 
-```bash
-kubectl delete -f ./kubernetes/istio-auth/istio-cluster-ca.yaml
-```
+   ```bash
+   kubectl get configmap istio -o yaml | grep authPolicy
+   ```
 
-#### Redeploying Istio and applications
+   Istio Auth is enabled if the line "authPolicy: MUTUAL\_TLS" is uncommented.
 
-Run the following command to uninstall Istio, and redeploy Istio without auth:
+2. Check the certificate and key files are mounted onto the application pod *app-pod*.
 
-```bash
-kubectl delete -f ./kubernetes/istio-auth-X.yaml
-kubectl apply -f ./kubernetes/istio-X.yaml
-```
+   ```bash
+   kubectl exec <app-pod> -c proxy -- ls /etc/certs
+   ```
 
-Also, redeploy your application by running:
+   ```bash
+   cert-chain.pem key.pem root-cert.pem
+   ```
 
-```bash
-kubectl replace -f <(istioctl kube-inject -f <your-app-spec>.yaml)
-```
+3. Check Istio Auth is enabled on Envoy proxies.
 
-#### Recovering the original config files
+   When Istio Auth is enabled for a pod, the *ssl_context* stanzas should be in the pod's proxy config.
+   The following commands verifies the proxy config on *app-pod* has *ssl_context* configured:
 
-The following command will recover the original *istio-auth-X.yaml* file.
+   ```bash
+   kubectl exec <app-pod> -c proxy -- ls /etc/envoy
+   ```
 
-```bash
-git checkout ./kubernetes/istio-auth-X.yaml
-```
+   The output should contain the config file "envoy-rev<X>.json". Use the file name in the following command:
 
-## Verifying Istio Auth setup
+   ```bash
+   kubectl exec <app-pod> -c proxy -- cat /etc/envoy/envoy-rev<X>.json | grep ssl_context
+   ```
 
-The following instructions assume the applications are deployed in the "default" namespace.
-They can be modified for deployments in a separate namespace.
+   If you see *ssl_context* lines in the output, the proxy has enabled Istio Auth.
 
-Verify AuthPolicy setting in ConfigMap:
+## Uninstalling Istio Auth
 
-```bash
-kubectl get configmap istio -o yaml | grep authPolicy
-```
-```bash
-# Istio Auth is enabled if the line "authPolicy: MUTUAL_TLS" is uncommented.
-```
+This section shows how to uninstall the cluster-level Istio CA and Istio that
+uses the cluster-level Istio CA.
 
-Check the certificate and key files are mounted onto the application pod *app-pod*:
+### Uninstalling the cluster-level Istio CA
+
+If you would like to disable auth for the entire cluster, you may want to remove the Istio CA.
+The following command removes Istio CA and its namespace *istio-system*.
 
 ```bash
-kubectl exec <app-pod> -c proxy -- ls /etc/certs
-```
-```bash
-# Expected files: cert-chain.pem, key.pem and root-cert.pem.
+kubectl delete -f install/kubernetes/templates/istio-auth/istio-cluster-ca.yaml
```

Comment:
They both need to be done.
This command removes the CA only. The following command removes other Istio components. I will make it clear.